src/com/facebook/buck/android/AndroidLibraryCompiler.java

```diff
@@ -0,0 +1,158 @@
+  }
+
+  @Override
+  public Iterable<BuildTarget> findDepsForTargetFromConstructorArgs(
+      BuildTarget buildTarget,
+      CellPathResolver cellRoots,
+      AndroidLibraryDescription.Arg constructorArg) {
+
+    return ImmutableList.of();
+  }
+}
+
+
+class JavaLibraryCompiler extends AndroidLibraryCompiler {
+
+  @Override
+  public CompileToJarStepFactory compileToJar(
+      AndroidLibraryDescription.Arg arg,
+      JavacOptions javacOptions,
+      BuildRuleResolver resolver) {
+
+    return new JavacToJarStepFactory(javacOptions, new BootClasspathAppender());
+  }
+}
+
+
+class ScalaAndroidLibraryCompiler extends AndroidLibraryCompiler {
+
+  private final ScalaBuckConfig scalaBuckConfig;
+  private Tool scalac;
+
+  ScalaAndroidLibraryCompiler(ScalaBuckConfig config) {
+    this.scalaBuckConfig = config;
+  }
+
+  private Tool getScalac(BuildRuleResolver resolver) {
+    if (scalac == null) {
+      scalac = scalaBuckConfig.getScalac(resolver);
+    }
+    return scalac;
+  }
+
+  @Override
+  public boolean trackClassUsage(JavacOptions javacOptions) {
+    return false;
+  }
+
+  @Override
+  public Iterable<BuildRule> getDeclaredDeps(AndroidLibraryDescription.Arg arg, BuildRuleResolver resolver) {
+    return ImmutableList.of(resolver.getRule(scalaBuckConfig.getScalaLibraryTarget()));
+  }
+
+  @Override
+  public Iterable<BuildRule> getExtraDeps(AndroidLibraryDescription.Arg arg, BuildRuleResolver resolver) {
+    return getScalac(resolver).getDeps(new SourcePathResolver(resolver));
+  }
+
+  @Override
+  public CompileToJarStepFactory compileToJar(AndroidLibraryDescription.Arg arg, JavacOptions javacOptions, BuildRuleResolver resolver) {
+    return new ScalacToJarStepFactory(
+        getScalac(resolver),
+        ImmutableList.<String>builder()
+            .addAll(scalaBuckConfig.getCompilerFlags())
+            .addAll(arg.extraArguments.get())
+            .addAll(Iterables.transform(
+                resolver.getAllRules(scalaBuckConfig.getCompilerPlugins()),
+                new Function<BuildRule, String>() {
+                  @Override
+                  public String apply(BuildRule input) {
+                    return "-Xplugin:" + input.getPathToOutput();
+                  }
+                }))
+            .build(),
+        new Function<BuildContext, Iterable<Path>>() {
+          @Override
+          public Iterable<Path> apply(BuildContext context) {
+            // TODO: can we access AndroidPlatformTarget directly instead of parsing the bootclasspath?
+            return Iterables.transform(
+                    Arrays.asList(context.getAndroidBootclasspathSupplier().get().split(File.pathSeparator)),
+                    new Function<String, Path>() {
+                      @Override
+                      public Path apply(String input) {
+                        return new File(input).toPath();
+                      }
+                    }
+                );
+          }
+        }
+    );
```

Comment:
Replaced android boot classpath supplier in `BuildContext` with `AndroidPlatformTarget` supplier. Seems better this way.